### DISCUSSION TITLE
Перерасчет таблиц циклов

### DISCUSSION THREAD

**Discussion by Spyker086** [totumonline/totum-ru-issues-and-discussions#403] [Как это сделать?] [answered]
Добрый день!

Подскажите как правильно поступить: есть таблицы рассчитываемые в цикле. Данные в строкой части таблиц загружаются sql запросами их внешней базы. Поэтому чтобы не инициировать запросы к внешней БД каждый раз при открытии таблиц стоят настройки - Исполнять код только при добавлении. Перерасчет планировали запускать рекалькулейтом задачей в крон ночью.

Но столкнулись с проблемой: каждый цикл перерассчитывается порядка минуты. Циклов около 300. Процесс перерассчета крон не выполняется до конца, а завершается через несколько минут. Подскажите как правильно решить такую задачу?

**Comment by erpic-ru** [comment, marked as answer]
Я очень часто использую этот способ: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/156

В вашем случае нужно только разовый самоотключаемый крон переделать в регулярный

**Reply by erpic-ru** [reply]
Самый простой способ - это второй крон, который будет включать ночью этот крон :) Количество пересчета ограничить 5-10 циклами или что там укладывается за минуту

**Reply by ttm-alexeypolunin** [reply]
Вообщем бы там посмотреть, как еще оптимизировать эти циклы, так-как расчет цикла = 1 минута — это очень много.

Если все упирается в то, что сторонняя база так долго отдает ответ, то тогда надо подгружать эти данные постоянно в фоне в какую-то промежуточную таблицу.
